### DISCUSSION TITLE
how to move DIFFUSERS_CACHE to new version

### DISCUSSION THREAD

**Discussion by LukeLIN-web** [huggingface/diffusers#9965] [Q&A] [answered]
ImportError: cannot import name 'DIFFUSERS_CACHE' from 'diffusers.utils' 

The old code is 
```python
from diffusers.utils import (
    CONFIG_NAME,
    DEPRECATED_REVISION_ARGS,
    DIFFUSERS_CACHE,
)
class DiffusionPipeline(ConfigMixin, PushToHubMixin):
    @classmethod
    def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.PathLike]], **kwargs):
        cache_dir = kwargs.pop("cache_dir", DIFFUSERS_CACHE)
            cached_folder = cls.download(
                pretrained_model_name_or_path,
                cache_dir=cache_dir,
```

How to modify the old code?

**Comment by asomoza** [comment]
Hi, it's not clear what you want to do, if it's old code it should work with the corresponding diffusers version, the newer versions don't use that constant anymore but you can just use the import for the `DiffusionPipeline` which is the base class for the pipelines.

If you want to learn how to convert the old code (which I don't really know why) you can just look at the `from_pretrained` method in the same class [here](https://github.com/huggingface/diffusers/blob/cd6ca9df2987c000b28e13b19bd4eec3ef3c914b/src/diffusers/pipelines/pipeline_utils.py#L503)

**Reply by LukeLIN-web** [reply]
> Hi, it's not clear what you want to do, if it's old code it should work with the corresponding diffusers version, the newer versions don't use that constant anymore but you can just use the import for the `DiffusionPipeline` which is the base class for the pipelines.
> 
> If you want to learn how to convert the old code (which I don't really know why) you can just look at the `from_pretrained` method in the same class [here](https://github.com/huggingface/diffusers/blob/cd6ca9df2987c000b28e13b19bd4eec3ef3c914b/src/diffusers/pipelines/pipeline_utils.py#L503)

Thank you for your help. I want to know what is `cache_dir` use for, whether we can remove it or whether this function is still using?

**Reply by asomoza** [reply, marked as answer]
`cache_dir` as the docstring says is:

> Path to a directory where a downloaded pretrained model configuration is cached if the standard cache is not used.

you can set it to `none` if you just want to use the default cache directory the `huggingface_hub` library uses.

**Reply by LukeLIN-web** [reply]
Thank you!
